src/pybind/mgr/dashboard/frontend/src/app/shared/api/auth.service.ts

```diff
@@ -21,15 +21,23 @@ export class AuthService {
     return this.http.post('api/auth/check', { token: token });
   }
 
-  login(credentials: Credentials) {
+  getNeedChangePassword(username: string) {
+    return this.http
+      .post('api/auth/getNeedChangePassword', {username: username})
```

Comment:
`getNeedChangePassword` does not follow REST conventions for naming Resources: the action `get` is stated by the HTTP method (`GET`) so no need to include it in the name, and regarding NeedChangePassword, it's rather a property of a User (the actual Resource here), than a resource by itself (`GET /api/auth/user/user_id`). However, I'm not sure if this issue comes from this very PR or it's already present in the design of this controller.